### ISSUE TITLE
[BUG] rke2 upgrade from 1.24 -> 1.25 `helm-install-calico-rke2` job is in `unavailable` state with `Upgrading rke2-calico Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "tigera-operator" namespace: "" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"`

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#40651]
**Rancher Server Setup**
- Rancher version: v2.7.2-rc3
- Installation option (Docker install/Helm Chart): helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): rke2, v1.24.9+rke2r1
- Proxy/Cert Details: valid certs 

**Information about the Cluster**
- Kubernetes version: v1.24.9+rke2r1 -> v1.25.6+rke2r1
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): rke2, aws node driver

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: standard user, cluster owner

**Describe the bug**

after upgrading the cluster, one job is `active`, but with a pod in an `unavailable` state. Viewing the logs of the pod shows that the upgrade of calico failed due to some missing PSPs 
```
Upgrading rke2-calico
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "tigera-operator" namespace: "" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"
```

**To Reproduce**

* deploy an rke2 cluster with dedicated nodes per role on v1.24.9+rke2r1
* wait for the cluster to become `active`
  * observe that the calico job on creation is successful
* upgrade the cluster to 1.25.6+rke2r1
* wait for the upgrade to succeed (and the cluster to get back to an `active` state
  * observe that the calico job upgrade does not complete

**Result**
 job is `active`, but with a pod in an `unavailable` state. Viewing the logs of the pod shows that the upgrade of calico failed due to some missing PSPs 
```
Upgrading rke2-calico
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "tigera-operator" namespace: "" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"
```

**Expected Result**

upgrade from 1.24 -> 1.25 should succeed

**Additional context**

* original use case involved deploying rke2 downstream cluster on 1.24, upgrading rancher (2.7.1 -> 2.7.2-rc3), then upgrading rke2 to 1.25. 
* same use case does not affect k3s
* same use case does not affect rke1

possibly related issue: https://github.com/rancher/rancher/issues/40510

**Comment by VestigeJ** [comment]
I've reproduced this several upgrades 

On the v1.24.9 r1 -> v1.25.6 upgrade

$ kubectl logs pod/helm-install-calico-trwu7
```
Upgrading rke2-calico
+ helm_v3 upgrade --set-string global.clusterCIDR=10.42.0.0/16 --set-string global.clusterCIDRv4=10.42.0.0/16 --set-string global.clusterDNS=10.43.0.10 --set-string global.clusterDomain=cluster.local --set-string global.rke2DataDir=/var/lib/rancher/rke2 --set-string global.serviceCIDR=10.43.0.0/16 rke2-calico /tmp/rke2-calico.tgz
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "tigera-operator" namespace: "" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"
ensure CRDs are installed first
+ exit
```

Upgrading v1.24.9+rk2r2 to v1.25.6+rke2r1

$ kubectl logs pod/helm-install-rke2-calico-wjhr2
```
Upgrading rke2-calico
+ shift 1
+ helm_v3 upgrade --set-string global.clusterCIDR=10.42.0.0/16 --set-string global.clusterCIDRv4=10.42.0.0/16 --set-string global.clusterDNS=10.43.0.10 --set-string global.clusterDomain=cluster.local --set-string global.rke2DataDir=/var/lib/rancher/rke2 --set-string global.serviceCIDR=10.43.0.0/16 rke2-calico /tmp/rke2-calico.tgz
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "tigera-operator" namespace: "" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"
ensure CRDs are installed first
+ exit
```

upgraded from the start v1.24.9+rke2r1 -> v1.24.9+rke2r2 -> v1.24.10+rke2r1 -> v1.25.6 still errors out

$ kubectl logs pod/helm-install-calico-sksxf
```
+ helm_v3 upgrade --set-string global.clusterCIDR=10.42.0.0/16 --set-string global.clusterCIDRv4=10.42.0.0/16 --set-string global.clusterDNS=10.43.0.10 --set-string global.clusterDomain=cluster.local --set-string global.rke2DataDir=/var/lib/rancher/rke2 --set-string global.serviceCIDR=10.43.0.0/16 rke2-calico /tmp/rke2-calico.tgz
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "tigera-operator" namespace: "" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"
ensure CRDs are installed first
+ exit
```

**Comment by VestigeJ** [comment]
on an upgrade from VERSION=v1.24.9+rke2r1 to COMMIT=1d6583a442106876b12fa784a2f783652b964f95

$ kubectl get pods -A

```
NAMESPACE         NAME                                                    READY   STATUS      RESTARTS        AGE
calico-system     calico-kube-controllers-687bc88ddf-x7pmm                1/1     Running     0               3m27s
calico-system     calico-node-278g6                                       1/1     Running     0               3m27s
calico-system     calico-node-6h9lm                                       1/1     Running     0               2m24s
calico-system     calico-node-hf6fb                                       1/1     Running     0               112s
calico-system     calico-node-js5nd                                       1/1     Running     0               2m56s
calico-system     calico-typha-7577db7b79-9trxm                           1/1     Running     0               3m29s
calico-system     calico-typha-7577db7b79-mfdpv                           1/1     Running     0               3m29s
default           akri-agent-daemonset-p6gl8                              1/1     Running     2 (5m46s ago)   12m
default           akri-agent-daemonset-qpslz                              1/1     Running     1 (5m46s ago)   12m
default           akri-agent-daemonset-vjc75                              1/1     Running     2 (4m14s ago)   12m
default           akri-agent-daemonset-xsgxc                              1/1     Running     4 (3m38s ago)   12m
default           akri-controller-deployment-78b9f74cc5-tjclw             1/1     Running     1 (6m23s ago)   12m
default           wasmcloud-wasmcloud-host-dc6d75fc9-d84dw                2/2     Running     3 (6m49s ago)   12m
kube-system       cloud-controller-manager-ip-172-31-19-47                1/1     Running     1 (4m25s ago)   13m
kube-system       cloud-controller-manager-ip-172-31-22-18                1/1     Running     3 (8m3s ago)    13m
kube-system       cloud-controller-manager-ip-172-31-22-188               1/1     Running     3 (5m41s ago)   19m
kube-system       etcd-ip-172-31-19-47                                    1/1     Running     1 (4m24s ago)   13m
kube-system       etcd-ip-172-31-22-18                                    1/1     Running     1 (8m25s ago)   13m
kube-system       etcd-ip-172-31-22-188                                   1/1     Running     1 (6m5s ago)    20m
kube-system       helm-install-rke2-calico-2mq7v                          0/1     Completed   0               4m1s
kube-system       helm-install-rke2-calico-crd-fxfgx                      0/1     Completed   0               4m1s
kube-system       helm-install-rke2-coredns-wd5hg                         0/1     Completed   0               4m1s
kube-system       helm-install-rke2-ingress-nginx-g95lp                   0/1     Completed   0               4m1s
kube-system       helm-install-rke2-metrics-server-9j8cr                  0/1     Completed   0               4m1s
kube-system       kube-apiserver-ip-172-31-19-47                          1/1     Running     1 (3m47s ago)   13m
kube-system       kube-apiserver-ip-172-31-22-18                          1/1     Running     1 (7m56s ago)   13m
kube-system       kube-apiserver-ip-172-31-22-188                         1/1     Running     1 (5m37s ago)   20m
kube-system       kube-controller-manager-ip-172-31-19-47                 1/1     Running     3 (4m3s ago)    13m
kube-system       kube-controller-manager-ip-172-31-22-18                 1/1     Running     3 (8m3s ago)    13m
kube-system       kube-controller-manager-ip-172-31-22-188                1/1     Running     3 (5m43s ago)   19m
kube-system       kube-proxy-ip-172-31-19-47                              1/1     Running     1 (3m26s ago)   13m
kube-system       kube-proxy-ip-172-31-22-18                              1/1     Running     1 (7m47s ago)   13m
kube-system       kube-proxy-ip-172-31-22-188                             1/1     Running     1 (5m23s ago)   19m
kube-system       kube-proxy-ip-172-31-23-250                             1/1     Running     1 (6m56s ago)   14m
kube-system       kube-scheduler-ip-172-31-19-47                          1/1     Running     1 (4m25s ago)   13m
kube-system       kube-scheduler-ip-172-31-22-18                          1/1     Running     1 (8m26s ago)   13m
kube-system       kube-scheduler-ip-172-31-22-188                         1/1     Running     1 (6m7s ago)    19m
kube-system       rke2-coredns-rke2-coredns-58fd75f64b-48xm8              1/1     Running     1 (5m26s ago)   20m
kube-system       rke2-coredns-rke2-coredns-58fd75f64b-gb29m              1/1     Running     1 (6m47s ago)   14m
kube-system       rke2-coredns-rke2-coredns-autoscaler-768bfc5985-q2x44   1/1     Running     1 (5m30s ago)   20m
kube-system       rke2-ingress-nginx-controller-k4bpk                     1/1     Running     1 (5m19s ago)   19m
kube-system       rke2-ingress-nginx-controller-s2js2                     1/1     Running     1 (4m4s ago)    12m
kube-system       rke2-ingress-nginx-controller-snz6z                     1/1     Running     1 (7m38s ago)   12m
kube-system       rke2-ingress-nginx-controller-xtcnv                     1/1     Running     1 (6m41s ago)   13m
kube-system       rke2-metrics-server-74f878b999-5x4ss                    1/1     Running     1 (5m30s ago)   19m
tigera-operator   tigera-operator-7bd6b54cb8-kkk99                        1/1     Running     0               3m54s
```


Upgrading from VERSION=v1.24.9+rke2r2 to COMMIT=1d6583a442106876b12fa784a2f783652b964f95

$ kubectl get pods -A
```
NAMESPACE         NAME                                                    READY   STATUS      RESTARTS        AGE
calico-system     calico-kube-controllers-687bc88ddf-qvxdx                1/1     Running     0               3m11s
calico-system     calico-node-5v9hq                                       1/1     Running     0               2m39s
calico-system     calico-node-c94bw                                       1/1     Running     0               2m14s
calico-system     calico-node-l9vhm                                       1/1     Running     0               3m11s
calico-system     calico-node-t4tbs                                       1/1     Running     0               111s
calico-system     calico-typha-6bfd5894c-5b57w                            1/1     Running     0               3m13s
calico-system     calico-typha-6bfd5894c-vt86v                            1/1     Running     0               3m13s
kube-system       cloud-controller-manager-ip-172-31-19-47                1/1     Running     6 (3m59s ago)   21m
kube-system       cloud-controller-manager-ip-172-31-22-18                1/1     Running     4 (6m8s ago)    21m
kube-system       cloud-controller-manager-ip-172-31-22-188               1/1     Running     4 (7m59s ago)   25m
kube-system       etcd-ip-172-31-19-47                                    1/1     Running     1 (4m28s ago)   21m
kube-system       etcd-ip-172-31-22-18                                    1/1     Running     1 (6m34s ago)   21m
kube-system       etcd-ip-172-31-22-188                                   1/1     Running     1 (8m16s ago)   25m
kube-system       helm-install-rke2-calico-5s5wl                          0/1     Completed   6               6m7s
kube-system       helm-install-rke2-calico-crd-97b6h                      0/1     Completed   0               6m7s
kube-system       helm-install-rke2-coredns-9z4ck                         0/1     Completed   0               6m7s
kube-system       helm-install-rke2-ingress-nginx-767cw                   0/1     Completed   0               6m7s
kube-system       helm-install-rke2-metrics-server-j4kht                  0/1     Completed   1               6m7s
kube-system       kube-apiserver-ip-172-31-19-47                          1/1     Running     2 (3m54s ago)   21m
kube-system       kube-apiserver-ip-172-31-22-18                          1/1     Running     2 (6m6s ago)    21m
kube-system       kube-apiserver-ip-172-31-22-188                         1/1     Running     2 (7m48s ago)   25m
kube-system       kube-controller-manager-ip-172-31-19-47                 1/1     Running     6 (4m10s ago)   21m
kube-system       kube-controller-manager-ip-172-31-22-18                 1/1     Running     7 (6m2s ago)    21m
kube-system       kube-controller-manager-ip-172-31-22-188                1/1     Running     6 (7m48s ago)   25m
kube-system       kube-proxy-ip-172-31-19-47                              1/1     Running     2 (3m41s ago)   21m
kube-system       kube-proxy-ip-172-31-22-18                              1/1     Running     2 (5m56s ago)   21m
kube-system       kube-proxy-ip-172-31-22-188                             1/1     Running     2 (7m34s ago)   25m
kube-system       kube-proxy-ip-172-31-23-250                             1/1     Running     2 (6m38s ago)   22m
kube-system       kube-scheduler-ip-172-31-19-47                          1/1     Running     2 (4m30s ago)   21m
kube-system       kube-scheduler-ip-172-31-22-18                          1/1     Running     2 (6m31s ago)   21m
kube-system       kube-scheduler-ip-172-31-22-188                         1/1     Running     2 (8m7s ago)    25m
kube-system       rke2-coredns-rke2-coredns-58fd75f64b-9z8jt              1/1     Running     1 (16m ago)     25m
kube-system       rke2-coredns-rke2-coredns-58fd75f64b-ll48w              1/1     Running     1 (13m ago)     22m
kube-system       rke2-coredns-rke2-coredns-autoscaler-768bfc5985-pzf7t   1/1     Running     1 (16m ago)     25m
kube-system       rke2-ingress-nginx-controller-4jblg                     1/1     Running     1 (13m ago)     22m
kube-system       rke2-ingress-nginx-controller-t2jx9                     1/1     Running     1 (10m ago)     21m
kube-system       rke2-ingress-nginx-controller-x7ttk                     1/1     Running     1 (12m ago)     21m
kube-system       rke2-ingress-nginx-controller-zj66r                     1/1     Running     1 (16m ago)     24m
kube-system       rke2-metrics-server-74f878b999-7b2w2                    1/1     Running     1 (16m ago)     24m
tigera-operator   tigera-operator-7bd6b54cb8-ssb8b                        1/1     Running     0               3m34s
```

Upgrading from VERSION=v1.24.10+rke2r1 to COMMIT=1d6583a442106876b12fa784a2f783652b964f95

$ kubectl get pods -A
```
NAMESPACE         NAME                                                    READY   STATUS      RESTARTS        AGE
calico-system     calico-kube-controllers-687bc88ddf-4lb4k                1/1     Running     0               4m9s
calico-system     calico-node-26l2b                                       1/1     Running     0               4m9s
calico-system     calico-node-7fllb                                       1/1     Running     0               3m37s
calico-system     calico-node-krj6v                                       1/1     Running     0               2m42s
calico-system     calico-node-qn62m                                       1/1     Running     0               3m6s
calico-system     calico-typha-5b587f88d-q4fxv                            1/1     Running     0               4m11s
calico-system     calico-typha-5b587f88d-vqslp                            1/1     Running     0               4m11s
default           akri-agent-daemonset-2f7d7                              1/1     Running     0               20m
default           akri-agent-daemonset-hwkmk                              1/1     Running     2 (5m13s ago)   20m
default           akri-agent-daemonset-j9hbh                              1/1     Running     1 (15m ago)     20m
default           akri-agent-daemonset-psqk8                              1/1     Running     1 (14m ago)     20m
default           akri-controller-deployment-78b9f74cc5-5knq9             1/1     Running     1 (4m56s ago)   20m
default           wasmcloud-wasmcloud-host-dc6d75fc9-dhwl9                2/2     Running     1 (20m ago)     20m
kube-system       cloud-controller-manager-ip-172-31-19-47                1/1     Running     4 (4m52s ago)   21m
kube-system       cloud-controller-manager-ip-172-31-22-18                1/1     Running     3 (15m ago)     21m
kube-system       cloud-controller-manager-ip-172-31-22-188               1/1     Running     3 (14m ago)     25m
kube-system       etcd-ip-172-31-19-47                                    1/1     Running     1 (5m42s ago)   21m
kube-system       etcd-ip-172-31-22-18                                    1/1     Running     1 (15m ago)     21m
kube-system       etcd-ip-172-31-22-188                                   1/1     Running     1 (15m ago)     24m
kube-system       helm-install-rke2-calico-crd-67hjb                      0/1     Completed   0               5m15s
kube-system       helm-install-rke2-calico-tqmsm                          0/1     Completed   0               5m15s
kube-system       helm-install-rke2-coredns-przcg                         0/1     Completed   0               5m15s
kube-system       helm-install-rke2-ingress-nginx-trptx                   0/1     Completed   0               5m15s
kube-system       helm-install-rke2-metrics-server-5fcgd                  0/1     Completed   0               5m15s
kube-system       kube-apiserver-ip-172-31-19-47                          1/1     Running     1 (4m56s ago)   21m
kube-system       kube-apiserver-ip-172-31-22-18                          1/1     Running     1 (15m ago)     21m
kube-system       kube-apiserver-ip-172-31-22-188                         1/1     Running     1 (14m ago)     25m
kube-system       kube-controller-manager-ip-172-31-19-47                 1/1     Running     3 (5m6s ago)    21m
kube-system       kube-controller-manager-ip-172-31-22-18                 1/1     Running     4 (15m ago)     21m
kube-system       kube-controller-manager-ip-172-31-22-188                1/1     Running     3 (14m ago)     25m
kube-system       kube-proxy-ip-172-31-19-47                              1/1     Running     1 (4m45s ago)   4m40s
kube-system       kube-proxy-ip-172-31-22-18                              1/1     Running     1 (15m ago)     21m
kube-system       kube-proxy-ip-172-31-22-188                             1/1     Running     1 (14m ago)     25m
kube-system       kube-proxy-ip-172-31-23-250                             1/1     Running     2 (11m ago)     22m
kube-system       kube-scheduler-ip-172-31-19-47                          1/1     Running     1 (5m32s ago)   21m
kube-system       kube-scheduler-ip-172-31-22-18                          1/1     Running     1 (15m ago)     21m
kube-system       kube-scheduler-ip-172-31-22-188                         1/1     Running     1 (15m ago)     25m
kube-system       rke2-coredns-rke2-coredns-58fd75f64b-hwdsr              1/1     Running     0               25m
kube-system       rke2-coredns-rke2-coredns-58fd75f64b-m48q4              1/1     Running     0               22m
kube-system       rke2-coredns-rke2-coredns-autoscaler-768bfc5985-sqnhk   1/1     Running     0               25m
kube-system       rke2-ingress-nginx-controller-9j8wr                     1/1     Running     0               21m
kube-system       rke2-ingress-nginx-controller-hkfmb                     1/1     Running     0               24m
kube-system       rke2-ingress-nginx-controller-p5r4b                     1/1     Running     0               21m
kube-system       rke2-ingress-nginx-controller-srdch                     1/1     Running     0               22m
kube-system       rke2-metrics-server-74f878b999-hzp7x                    1/1     Running     0               24m
tigera-operator   tigera-operator-7bd6b54cb8-78sth                        1/1     Running     0               4m33s
```

**Comment by jameson-mcghee** [comment]
I have reproduced this issue on `v2.7-head` (Commit ID: 6cca23f), 

**Helm Pod Log with Error:**
[helm-install-rke2-calico-gmscn_helm.log](https://github.com/rancher/rancher/files/10886714/helm-install-rke2-calico-gmscn_helm.log)


![image](https://user-images.githubusercontent.com/13294321/222844669-767ed671-79fc-495d-811d-c9631ff5b183.png)
![image](https://user-images.githubusercontent.com/13294321/222844678-b19b709f-0a00-4ead-b317-018bfe5d3c8d.png)

**Comment by brandond** [comment]
This has been fixed in RKE2 in the February releases, as referenced from the RKE2 issues linked above. We will need to test upgrading to the RCs (or GA releases once they are available). Upgrading from 1.24 to to any prior 1.25 or 1.26 release will see this issue.

**Comment by snasovich** [comment]
Moving "To Test" as February patched got merged into 2.7 KDM per https://github.com/rancher/kontainer-driver-metadata/pull/1099.
Please validate on the upgrades from `v1.24.11+rke2r1` to `v1.25.7+rke2r1`.

**Comment by slickwarren** [comment]
tested on v2.7-head (a49d36c):
* upgrading rke2 from `v1.24.11+rke2r1` to `v1.25.7+rke2r1` -- pass
